### PR TITLE
Add the shareable cart feature

### PR DESCRIPTION
**Pull request by dreampulse**

### CODE REVIEW

**Comment by dreampulse** [comment]
I think it would be valuable to add a cypress test here @webholics what do you think?

File: src/app/reducer/core.ts [hunk 1]
```diff
@@ -849,6 +849,73 @@ const loadConfiguration = (
...
+  }
+
+  const modelConfigs: ModelConfig[] = zip(
+    payload.models,
+    payload.quotes,
+    payload.shippings,
+    payload.modelConfigIds
+  ).map(([model, quote, shipping, modelConfigId]) => ({
+    type: 'UPLOADED',
+    quantity: (quote as Quote).quantity,
```
Comment (dreampulse) [inline src/app/reducer/core.ts]: I've fount out. That the backend always seems to return here `1`.
@alex-all3dp & @henry-all3dp

Comment (alex-all3dp) [reply src/app/reducer/core.ts]: Can't reproduce this. Quantity is correct (larger than 1 if so selected in the original cart) and displayed correctly in the client as well?

**Comment by alex-all3dp** [comment]
@dreampulse @Dansken When navigating back from a restored cart to the upload page, the restored items are not displayed in the selection. Could be confusing to the user?

**Comment by Dansken** [comment]
@dreampulse Hi Jonathan,

Could I ask you to include a campaign tracking code that is "shareable_cart" so that we can track orders coming in via the share cart feature?

File: src/app/reducer/core.ts (hunk 1 above)
Comment (dreampulse) [reply src/app/reducer/core.ts]: Hmm.. now it seems to work

**Comment by dreampulse** [comment]
> @dreampulse @Dansken When navigating back from a restored cart to the upload page, the restored items are not displayed in the selection. Could be confusing to the user?

This is the default behaviour right now. If you want to change that I suggest to do that in a separate PR

**Comment by dreampulse** [comment]
> Could I ask you to include a campaign tracking code that is "shareable_cart" so that we can track orders coming in via the share cart feature?

Ok i'll add `?utm_campaign=shareable_cart` to the shared link

File: src/app/reducer/core.ts [hunk 2]
```diff
@@ -849,6 +849,74 @@ const loadConfiguration = (
...
+    })
+  )
+
+const offerReceived = (
+  state: CoreState,
+  {payload}: cartActions.OfferReceivedAction
+): CoreReducer => {
+  const arrayLengths = payload.models.length
+
+  // TODO: This might change and we need to match the items using the modelId
```
Comment (dreampulse) [inline src/app/reducer/core.ts]: Eventuell ändert sich hier noch die API

File: src/app/component/paypal-button/paypal-button.js [hunk 3]
```diff
@@ -24,8 +24,7 @@ class PaypalButton extends Component {
         height: BUTTON_HEIGHT
       }
     }
-
-    global.paypal.Button.render(options, this.paypalButton)
+    global.paypal && global.paypal.Button && global.paypal.Button.render(options, this.paypalButton)
```
Comment (webholics) [inline src/app/component/paypal-button/paypal-button.js]: hier müsste man jetz eigentlich so lange retrys machen bis der paypal button da is. kannst gern einfach n ticket für anlegen und hier ein todo reinsetzen. aber das muss gemacht werden.

File: src/app/container/main-app.js [hunk 4]
```diff
@@ -59,6 +66,7 @@ const MainApp = ({initDone, initTriggered, initAction}) => {
         <Route component={CartPage} path="/cart" exact />
         <Route component={SuccessPage} path="/success" exact />
         <Route component={ConfigurationPage} path="/configuration/:id" exact />
+        <Route component={SharedCartPage} path="/cart/:id" exact />
```
Comment (webholics) [inline src/app/container/main-app.js]: OfferPage und "/offer/:id"

File: src/app/container/modal/share-cart.js [hunk 5]
```diff
@@ -52,6 +52,6 @@ export default compose(
     mapDispatchToProps
   ),
   withProps(({cartId}) => ({
-    cartUrl: `${global.location.origin}/cart/${cartId}`
+    cartUrl: `${global.location.origin}/cart/${cartId}?utm_campaign=shareable_cart`
```
Comment (webholics) [inline src/app/container/modal/share-cart.js]: müssten wir dann auch umbennenen zu `/offer/${encodeURIComponent(offerId)}`

**Pull request merged**